srml/treasury/src/lib.rs

```diff
@@ -325,21 +335,31 @@ impl<T: Trait> Module<T> {
 		Self::deposit_event(RawEvent::Rollover(budget_remaining));
 	}
 
+	/// Return the amount of money in the pot.
 	fn pot() -> BalanceOf<T> {
 		T::Currency::free_balance(&Self::account_id())
+			// Must never be less than 0 but better be safe.
+			.saturating_sub(T::Currency::minimum_balance())
 	}
 }
 
 impl<T: Trait> OnUnbalanced<NegativeImbalanceOf<T>> for Module<T> {
 	fn on_unbalanced(amount: NegativeImbalanceOf<T>) {
-		T::Currency::resolve_creating(&Self::account_id(), amount);
+		// Must resolve into existing but better be safe.
+		let _ = T::Currency::resolve_creating(&Self::account_id(), amount);
```

Comment:
Now that we have better runtime logging I do get tempted to do something like `resolve_exisiting().unwrap_or_else(|_| { log("CODE RED, SOS"); resolve_creating() )`. 

But I assume it depends on the proof. If it is logically impossible to cause account creation then this is also good.

Reply:
interesting indeed a lot of such safe operation could be logged 🤔